### ISSUE TITLE
Sandboxing command line applications

### ISSUE THREAD

**Issue by benjaminwinger** [sandboxie-plus/Sandboxie#53]
I use sandboxie for sandboxing command line applications in a non-interactive scripting environment, however the issue is that currently sandboxie doesn't produce any stdout or stderr when running command line applications, so this limits what can be achieved, and makes debugging difficult. 

Would it be possible to have sandboxie's start command produce stdout and stderr from its child process? 

If memory serves Windoes treats gui and cli programs separately, so maybe this would require a `start-cli.exe` variant for command line applications (I'm primarily a Linux developer and am using sandboxie to provide compatibility to Windows users).

It would also be useful if the sandbox configuration could be passed as command line arguments to `start.exe` so that Sandboxie could be run in a stateless configuration. Currently I'm creating and deleting a sandbox before and after each command, which is unnecessary and a little slow.

**Comment by DavidXanatos** [comment]
Emmm.... that is in fact far more difficult that it seams, as the processes are started using the SbieSvc and it may violate the box isolation i.e. it would require some IPC white listing in the sandboxie.ini

**Comment by benjaminwinger** [comment]
Why is that? I'm not too familiar with the architecture of sandboxie, but I have noticed that it differs significantly from the sandboxing tools I use on other platforms ([bubblewrap](https://github.com/projectatomic/bubblewrap) on Linux, sandbox-exec on macOS). 

Is the need to be started using the service something to do with the way sandboxie redirects file i/o to its sandboxed directory tree? I'm used to sandbox tools which either allow or block i/o, without having a mechanism for "pretending" it succeeds, with the idea that you would generally pass arguments to the programs being run to indicate the directory they should store files in anyway (generally a temporary directory, where giving direct access to the filesystem isn't problematic). FireJail on the other hand apparently supports using an overlayfs to have behaviour similar to what sandboxie does, but there the filesystem driver would handle maintaining the illusion of the normal filesystem, and a chroot is used instead of redirecting i/o operations in userspace, but I digress.

If I'm not completely misunderstanding things, would it be possible to have a version of start.exe that could execute the subprocess directly, even at the cost of certain features which aren't critical to security?

**Comment by DavidXanatos** [comment]
The Sandboxie driver can force a program started arbitrary into a sandbox.
I would suggest to leverage this mechanism for your use case. 

ForceProcess=myConsoleProxy.exe

Create a proxy exe that starts starts your target process and is itself configured to run be forced into a sandbox, as this forced it child processes as well.
And make the exe to relay all the console from its child to its own output which you should be able to monitor.

I haven't tested it but it should work.

**Comment by benjaminwinger** [comment]
I've tested it, and it does work. It's a little unintuitive, but it gets the job done, thanks.

That being said, I had issues integrating this solution, mostly related to the fact that ForceProcess works on binaries and not scripts, that the software I'm using it in is a python package, and that packaging binary executables is not something python's packager is designed for. I eventually got it working, and it's better than nothing at the moment, but it would be great to have a cleaner solution eventually.
